helsingborg-stad/app-mitt-helsingborg
# Feature/booking form

## Explain the changes you’ve made

I have added the ability to book a meeting through the app, as well as retrieving existing meetings from the backend.

## Explain why these changes are made

These changes are necessary in order to give the user of the app the ability to book a meeting and see their existing meetings.

## Explain your solution

In order to implement this feature, I have added the following new screens:
```
screens/BookingFormScreen.tsx
containers/Form/BookingForm.tsx
```
To accommodate the needs of these screens and to centralize the functions used in the booking service, I have created the helper file `helpers/BookingHelper.ts` for helper functions related to booking, and `types/BookingTypes.ts` to contain the types used in the booking system.
In addition, I have updated several existing files to make use of the new types, and solved a few minor bugs related to the behaviour of the components used in the new booking form screens.

## How to test the changes?

First make sure the backend is set up correctly:
1. Deploy the latest version of `booking`, `bookables` and `timeslot` services from the `sls-api` and `sls-resources` respositories.
2. Deploy the correct `datatorgetEnvs` and `bookablesEnvs` parameters to the backend.
3. Make sure you have access to the environment that datatorget is connected to, and that you have a user in a shared mailbox group that is defined in one of the services described in `bookablesEnvs`.
4. Create a booking in the Outlook calendar of the user with the exact title "MH bokningsbar tid".

Now you can launch the app to test the changes:
1. Checkout this branch
2. Make sure the env is not in storybook mode
4. Fire upp the simulator by running the command `yarn ios`
5. Navigate to the calendar screen
6. Test the new functionalities by pressing "Boka möte"

## Was this feature tested in the following environments?
- [] Storybook on a iOS device/simulator.
